pkg/cmd/properties/properties.go

```diff
@@ -179,15 +203,45 @@ func (p *properties) AddStringSliceProperty(name string, defaultVal []string, de
 	}
 }
 
-func (p *properties) AddDurationProperty(name string, defaultVal time.Duration, description string) {
+func (p *properties) AddDurationProperty(name string, defaultVal time.Duration, description string, options ...DurationOpt) {
 	if p.rootCmd != nil {
 		flagName := p.nameToFlagName(name)
+
+		// set up range
+		lowerLimit := fmt.Sprintf(lowerLimitName, flagName)
+		upperLimit := fmt.Sprintf(upperLimitName, flagName)
+
+		// validate if WithLowerLimit and WithUpperLimit were called
+		for _, option := range options {
+			option(p)
+		}
+
+		// make sure limits are valid
+		if p.addDurationRange(p.lowerLimit, p.upperLimit) {
+			// add range value
+			p.rootCmd.Flags().Duration(lowerLimit, p.lowerLimit, "the value entered is lower than the supported lower limit for this configuration")
+			p.rootCmd.Flags().Duration(upperLimit, p.upperLimit, "the value entered is higher than the supported higher limit for this configuration")
+		}
+
 		p.rootCmd.Flags().Duration(flagName, defaultVal, description)
 		p.bindOrPanic(name, p.rootCmd.Flags().Lookup(flagName))
 		p.rootCmd.Flags().MarkHidden(flagName)
 	}
 }
 
+func (p *properties) addDurationRange(lowerLimitVal, upperLimitVal time.Duration) bool {
+	// if either range value == 0, return false
+	if lowerLimitVal == 0 || upperLimitVal == 0 {
```

Comment:
So this requires each limit to be set to use either? Shouldn't we allow for floor, ceiling, or both?

Reply:
@vivekschauhan

Reply:
I like the proposal,  makes sense to me. 👍🏾. Thoughts,  @vivekschauhan, I can get with this.

Reply:
@vivekschauhan agreed with this

Reply:
okee dokee